src/v/cloud_storage/remote_segment.cc

```diff
@@ -881,17 +910,91 @@ ss::future<> remote_segment::run_hydrate_bg() {
     _hydration_loop_running = false;
 }
 
-namespace {
+ss::future<> remote_segment::service_chunk_requests() {
+    fragmented_vector<ss::future<ss::file>> chunk_op_results;
+    chunk_op_results.reserve(_chunk_waiters.size());
+
```

Comment:
What's the limiting factor for the growth of this vector? Wondering if it makes sense to batch the downloads at all, rather than doing them all at once, but maybe it's fine if this doesn't grow much

Reply:
The upper limit here is quite small, at maximum this will be the number of chunks in the segment, so around 8 given our default sizes, if there are consumers requesting all chunks in the segment at the same time (or aggressive prefetch).

It does make sense to batch the downloads to guard against corner cases, I can add this in the next PR which will make changes to the prefetch behavior.